socks5.py

```diff
@@ -402,8 +508,37 @@ def run_wpad_server(server):
     except KeyboardInterrupt:
         pass
 
+def print_traffic_info():
+    global inbound_traffic
+    global outbound_traffic
+    while True:
+        time.sleep(30)
+        with traffic_lock:
+            inbound_mbps = (inbound_traffic * 8) / (30 * 1024 * 1024)
+            outbound_mbps = (outbound_traffic * 8) / (30 * 1024 * 1024)
+            inbound_traffic = 0
+            outbound_traffic = 0
+        # Clear the console
+        if 'Pythonista' in sys.executable:
+            console.clear()
+        else:
+            print('\033c', end='')
+# print initial_output
+        print(initial_output)
+        print("PAC URL: http://{}:{}/wpad.dat".format(PROXY_HOST, WPAD_PORT))
+        print("SOCKS Address: {}:{}".format(PROXY_HOST or SOCKS_HOST, SOCKS_PORT))
+
+        # Print the table
+        print(f"{'Direction':<10} | {'Traffic (Mbps)':<15}")
+        print(f"{'-'*10} | {'-'*15}")
+        print(f"{'Inbound':<10} | {inbound_mbps:<15.2f}")
```

Comment:
I love this idea, thanks! It's certainly a lot more useful than the logspew from before.

Some thoughts to make it more useful:
- Can the *total* amount of traffic be displayed? This will be very useful for people on metered connections (i.e. any cell connection with a bandwidth cap)
- 30 seconds per update seems a bit infrequent. Faster updates with a sliding window average of speed would be more useful. (My go-to favorite is an EWMA filter, which is trivially simple to implement and requires minimal storage)

Reply:
Will fix both in the next commit.

And yes, the logspew from before was pretty terrible. I've cleaned up most of it, but it's still not perfect. There are also some race conditions which can cause Tracebacks to appear if you close a connection very quickly after attempting it before the server accepts it.

I'd like to add some more pretty outputs here later on to be able to see what hosts we are connected to, but will probably focus on that after improving the 'happy eyeballs' ipv4/ipv6 connection feature.